structuralcodes/sections/section_integrators/_marin_integrator.py

```diff
@@ -126,29 +132,92 @@ def get_input_polygon(polygon, coeffs):
 
                     if isinstance(result, Polygon):
                         # If the result is a single polygon
-                        get_input_polygon(result, coeffs[p])
+                        self._get_input_polygon(result, coeffs[p], input)
                     elif isinstance(result, MultiPolygon):
                         # If the result is a MultiPolygon
                         for polygon in result.geoms:
-                            get_input_polygon(polygon, coeffs[p])
+                            self._get_input_polygon(polygon, coeffs[p], input)
+
+    def _process_point_geometries(
+        self,
+        geo: CompoundGeometry,
+        strain: ArrayLike,
+        input: t.List,
+        integrate: t.Literal['stress', 'modulus'] = 'stress',
+    ):
+        """Process Point geometries filling the input data."""
         # Tentative proposal for managing reinforcement (PointGeometry)
         x = []
         y = []
-        F = []
-        for pg in rotated_geom.point_geometries:
+        IA = []
+        for pg in geo.point_geometries:
             xp, yp = pg._point.coords.xy
             xp = xp[0]
             yp = yp[0]
             A = pg.area
-            strain = strain_rotated[0] + strain_rotated[1] * yp
+            strain_ = strain[0] + strain[1] * yp
             x.append(xp)
             y.append(yp)
-            F.append(pg.material.get_stress(strain) * A)
-        prepared_input.append((1, np.array(x), np.array(y), np.array(F)))
+            if integrate == 'stress':
+                IA.append(pg.material.get_stress(strain_) * A)
+            elif integrate == 'modulus':
+                IA.append(pg.material.get_tangent(strain_) * A)
+        input.append((1, np.array(x), np.array(y), np.array(IA)))
+
+    def prepare_input(
+        self,
+        geo: CompoundGeometry,
+        strain: ArrayLike,
+        integrate: t.Literal['stress', 'modulus'] = 'stress',
+    ) -> t.Tuple[float, t.Tuple[np.ndarray, np.ndarray, np.ndarray]]:
+        """Prepare general input to the stress integration.
+
+        Calculate the stresses based on strains in a set of points.
```

Comment:
Suggest to modify the docstring similar to the suggestions above.

Reply:
done